File: Sistemas Operativos_Octavio/Sistemas Operativos/Laboratorio/labsisops/upsc1/FechaG.cc
// FechaG.cc

#include <stdio.h>
#include <time.h>
#include <unistd.h>

time_t a,b;
struct tm *vtm;

void aux()
{
    printf( "HOUR= %d      ", vtm->tm_hour );
    printf( "MINUTE= %d      ", vtm->tm_min );
    printf( "SECOND= %d\n", vtm->tm_sec);
    printf( "DAY OF MONTH= %d      ", vtm->tm_mday);
    printf( "MONTH= %d      ", vtm->tm_mon + 1);
    printf( "YEAR= %d\n", vtm->tm_year + 1900);
    printf( "WEEK DAY= %d      ", vtm->tm_wday);
    printf( "YEAR DAY= %d\n", vtm->tm_yday + 1);
    return;
}

int main()
{
    char *st1;

    time( &a );
    printf( "a= %ld seg.\n",a);

    // Similar al comando "date"
    st1 = ctime( &a );
    printf( "st1= %s\n", st1);

    vtm = localtime( &a );
    aux();

    printf( "------------------------------\n");
    sleep(4);
    printf( "------------------------------\n");

    time( &b );
    printf( "b= %ld seg.\n", b);

    // Similar al comando "date"
    st1 = ctime( &a );
    printf( "st1= %s\n", st1);

    vtm = localtime( &a );
    aux();

    printf("La diferencia es de %ld segundos. \n",b-a);

    return 0;
}

File: Sistemas Operativos_Octavio/Sistemas Operativos/Laboratorio/labsisops/upsc1/FechaG.java
import java.util.*;

public class FechaG
{
   public static Calendar calendar;

   public static void main( String args[] )
   {
      long t1 = System.currentTimeMillis();
      calendar = new GregorianCalendar();
      System.out.println("HOY= " + (new Date()).toString() );
      aux();

      System.out.println("---------------------------");
      try {  
        Thread.sleep(4000); 
      }
      catch (InterruptedException e) { 
      } 
      System.out.println("---------------------------");

      calendar = new GregorianCalendar();
      System.out.println("HOY= " + (new Date()).toString() );
      aux();
      long t2 = System.currentTimeMillis();
      t2 = t2-t1;

      System.out.println("La diferencia es de "+t2+" milisegundos.");

   } // main

   public static void aux()
   {
      System.out.print("DATE: " + calendar.get(Calendar.DATE) +
                       "        ");
      System.out.print("MONTH: " + (calendar.get(Calendar.MONTH)+1) +
                       "        ");
      System.out.println("YEAR: " + calendar.get(Calendar.YEAR));
      System.out.print("WEEK_OF_YEAR: " + calendar.get(Calendar.WEEK_OF_YEAR) +
                       "        ");
      System.out.println("WEEK_OF_MONTH: " + calendar.get(Calendar.WEEK_OF_MONTH));
      System.out.print("DAY_OF_MONTH: " + calendar.get(Calendar.DAY_OF_MONTH) +
                       "        ");
      System.out.print("DAY_OF_YEAR: " + calendar.get(Calendar.DAY_OF_YEAR) +
                       "        ");
      System.out.println("DAY_OF_WEEK: " +calendar.get(Calendar.DAY_OF_WEEK));
      System.out.print("HOUR: " + calendar.get(Calendar.HOUR) +
                       "        ");
      System.out.println("AM_PM: " + calendar.get(Calendar.AM_PM));
      System.out.println("HOUR_OF_DAY: " + calendar.get(Calendar.HOUR_OF_DAY));
      System.out.print("MINUTE: " + calendar.get(Calendar.MINUTE) +
                       "        ");
      System.out.print("SECOND: " + calendar.get(Calendar.SECOND) +
                       "        ");
      System.out.println("MILLISECOND: " + calendar.get(Calendar.MILLISECOND));
   }// aux
} // FechaG

